python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -525,6 +559,7 @@ def on_task_output_generated(self, task_index: int, output: RefBundle):
         self.bytes_task_outputs_generated += output_bytes
 
         task_info = self._running_tasks[task_index]
+        # TODO(mowen): Shouldn't this be `if task_info.num_outputs != 0`?
```

Comment:
No. `num_outputs == 0` means this is the first time we receive an output from this task, thus bump `num_tasks_have_outputs`.